### PR TITLE
Python-SDK-implementation-for-RightSizing-Rules-CRUD,-attach-and-detach

### PR DESCRIPTION
**Pull request by adityas1netapp**
_This is the default pull request template. You can customize it by adding a `pull_request_template.md` at the root of your repo or inside the `.github` folder._

# Jira Ticket: [SPOTAUT-18749](https://spotinst.atlassian.net/browse/SPOTAUT-18749)

_Include a link to your Jira Ticket_
_Example: [JIRAISS-1234](https://spotinst.atlassian.net/browse/JIRAISS-1234)_

# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [ ] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have validated all the requirements in the Jira task were answered
- [ ] I have all neccessary approvals for the design/mini design of this task
- [ ] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**) 

[SPOTAUT-18749]: https://spotinst.atlassian.net/browse/SPOTAUT-18749?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: spotinst_sdk2/models/ocean/rightsizing/__init__.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    interval_months: List[int]
+    week_of_the_month: List[str]
+    weekly_repetition_basis: WeeklyRepetitionBasis
+    interval_hours: IntervalHours
+    """
+
+    def __init__(
+            self,
+            interval_months: List[int] = none,
+            week_of_the_month: Enum = none,
```
Comment (anuragsharma-123) [inline spotinst_sdk2/models/ocean/rightsizing/__init__.py]: This is enum or list enum? Please check how enums are defined in other classes with all possible values and define similarly.

File: spotinst_sdk2/__init__.py [hunk 2]
```diff
@@ -53,6 +53,8 @@ def client(self, service, print_output=True, log_level=None, user_agent=None, ti
                                               user_agent=user_agent, timeout=timeout),
             "ocean_aws": OceanAwsClient(session=self.session, print_output=print_output, log_level=log_level,
                                         user_agent=user_agent, timeout=timeout),
+            "ocean_aws_rightsizing": OceanRightSizingClient(session=self.session, print_output=print_output, log_level=log_level,
```
Comment (anuragsharma-123) [inline spotinst_sdk2/__init__.py]: "ocean_aws_rightsizing" should be changed to "ocean_rightsizing"

Comment (adityas1netapp) [reply spotinst_sdk2/__init__.py]: changed

File: spotinst_sdk2/models/ocean/rightsizing/__init__.py (hunk 1 above)
Comment (adityas1netapp) [reply spotinst_sdk2/models/ocean/rightsizing/__init__.py]: updated

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**